### PR TITLE
Builds and Artifacts API

### PR DESCRIPTION
**Pull request by ejholmes**
This adds a Builds and Artifacts API so that external tools (like Empire) can trigger builds on demand and obtain the artifacts that result from those builds.

I've split this into two commits:

1. First commit contains a spec for the API using json-schema and auto generates docs and a Go client from it.
2. Second commit adds an implementation in `server/api`.

**TODO**

* Auth.
* Use github.com/pkg/httpx
* Add a default transport to the generated client to handle error resources.
* Move slack.branchResolver into the Conveyor struct, so sha can be optional in the builds API.
* Make branch an optional parameter when creating a new build?

### CODE REVIEW

File: client/conveyor/conveyor.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+		}
+	}
+	if lr.Descending {
+		hdrval += ", order=desc"
+	}
+	req.Header.Set("Range", hdrval)
+	return
+}
+
+// Bool allocates a new int value returns a pointer to it.
```
Comment (phobologic) [inline client/conveyor/conveyor.go]: copy-pasta?

Comment (ejholmes) [inline client/conveyor/conveyor.go]: autogenerated pasta :)

**Comment by phobologic** [comment]
:+1:

**Comment by ejholmes** [comment]
Note to self. We should error out of `Conveyor.Build` if the GitHub repository hasn't been enabled, otherwise we could potentially build repos that haven't been explicitly enabled (e.g. a user makes a typo when using the `/conveyor build` slash command).

**Pull request merged**